### PR TITLE
Enforce rebuilding of project in build.fsx

### PR DESCRIPTION
**Pull request by heynickc**
An attempt to fix intermittent integration test failures for the tests added to validate that assemblies with dependencies are loaded into the runner.  Adds flag `--no-incremental` which is synonymous with "Rebuild".

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@heynickc started failing on Mono this time :p

File: tests/NBench.Tests.End2End/NBenchIntegrationTest.WithDependencies.cs [hunk 1]
```diff
@@ -77,5 +77,13 @@ private static TestPackage LoadPackageWithDependencies(IEnumerable<string> inclu
             package.Validate();
             return package;
         }
+
+        public void Dispose()
+        {
+            if (_discovery != null)
+            {
+                _discovery = null;
```
Comment (Aaronontheweb) [inline tests/NBench.Tests.End2End/NBenchIntegrationTest.WithDependencies.cs]: Why is this necessary exactly?

File: build.fsx [hunk 2]
```diff
@@ -77,14 +77,27 @@ Target "Build" (fun _ ->
             (fun p -> 
                 { p with
                     Project = project
-                    Configuration = configuration })   
+                    Configuration = configuration 
+                    AdditionalArgs = ["--no-incremental"]}) // "Rebuild"
```
Comment (Aaronontheweb) [inline build.fsx]: Makes sense

**Comment by heynickc** [comment]
@Aaronontheweb no it wasn't strictly .NET Core in terms of being able to load assemblies that were referenced physically.  Both frameworks were given a fall back to look in the bin directory for missing assemblies when loading.

File: tests/NBench.Tests.End2End/NBenchIntegrationTest.WithDependencies.cs (hunk 1 above)
Comment (heynickc) [inline tests/NBench.Tests.End2End/NBenchIntegrationTest.WithDependencies.cs]: This forces teardown of the `ReflectionDiscovery` class on every test method run by xUnit.  The behavior was pointing to test failures when running both tests in this class back-to-back.  This was mostly a check to make sure there weren't any remnants from the `ShouldPassAllBenchmarks` assembly loading process during that test.  Running the test `LoadAssemblyCorrect` in isolation passed 100% of the time, but when running in CI the behavior was flaky.
